homeassistant/components/home_connect/__init__.py

```diff
@@ -138,11 +138,18 @@ async def _async_service_program(call, method):
         """Execute calls to services taking a program."""
         program = call.data[ATTR_PROGRAM]
         device_id = call.data[ATTR_DEVICE_ID]
-        options = {
-            ATTR_KEY: call.data.get(ATTR_KEY),
-            ATTR_VALUE: call.data.get(ATTR_VALUE),
-            ATTR_UNIT: call.data.get(ATTR_UNIT),
-        }
+
+        options = []
+
+        option_key = call.data.get(ATTR_KEY)
+        if option_key is not None:
+            option = {ATTR_KEY: option_key, ATTR_VALUE: call.data.get(ATTR_VALUE)}
+
+            option_unit = call.data.get(ATTR_UNIT)
+            if option_unit is not None:
+                setattr(option, ATTR_UNIT, option_unit)
```

Comment:
This is not how you set an item in a dict in Python. There's no reason to use `setattr` here.
```suggestion
                option[ATTR_UNIT] = option_unit
```

Reply:
Changed, thanks, python is not my first language